### DISCUSSION TITLE
How to use zustand instead of useState to manage the state of a single component?

### DISCUSSION THREAD

**Discussion by vaynevayne** [pmndrs/zustand#1760] [Q&A] [answered]
I want to use zustand instead of useState because I need to use its middleware, persistence and immer and undo redo,
I don't need multiple components to share state.

```jsx
// before 
const Demo=()=>{
const [form,setForm] = useState({})
}

// after
import useZustand from './useZustand'
const Demo=()=>{
const [form,setForm] = useZustand({}) // i don't know how to implement it 
}

```

**Comment by JacobWeisenburger** [comment, marked as answer]
Using zustand for 1 component is the same as using it for many components. Have you had a chance to read the readme?

**Reply by vaynevayne** [reply]
how to use it for 1 component  ? I didn't see it in readme. Are there any best practices? thanks
